File: backend/RestaurantService/src/main/java/com/fooddelivery/restaurant/model/Restaurant.java
package com.fooddelivery.restaurant.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Restaurant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String address;
    private String contactNumber;
    private Integer ownerId; // ID of the user with RESTAURANT_OWNER role
    private String cuisine;
    private Double rating;

    private String img;
    private String status; // PENDING, APPROVED, REJECTED

    @OneToMany(mappedBy = "restaurant", cascade = CascadeType.ALL)
    private List<MenuItem> menu;
}

File: backend/AdminService/Models/Restaurant.cs
namespace AdminService.Models
{
    public class Restaurant
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Address { get; set; }
        public required string ContactNumber { get; set; }
        public int OwnerId { get; set; }
        public required string Cuisine { get; set; }
        public double Rating { get; set; }
        public required string Img { get; set; }
    }
}
